src/main/kotlin/app/revanced/patches/tiktok/interaction/downloads/patch/DownloadsPatch.kt

```diff
@@ -50,7 +60,76 @@ class DownloadsPatch : BytecodePatch(
+            if (instruction.opcode != Opcode.CONST_STRING) continue
+            val reference = (instruction as ReferenceInstruction).reference as StringReference
+            if (reference.string != "video/mp4") continue
+            val targetInstruction = instructions[index + 1]
+            if (targetInstruction.opcode != Opcode.INVOKE_STATIC) continue
+            targetOffset = index + 1
+            break
+        }
+        if (targetOffset == -1) return PatchResultError("Can not find download path uri method.")
+        //Change videos' download path.
+        val downloadPath = "$downloadPathParent/$downloadPathChild"
+        val downloadUriMethod = data
+            .toMethodWalker(DownloadPathParentFingerprint.result!!.method)
+            .nextMethod(targetOffset, true)
+            .getMethod() as MutableMethod
+        downloadUriMethod.implementation!!.instructions.forEachIndexed { index, instruction ->
+            if (instruction.opcode == Opcode.SGET_OBJECT) {
+                val overrideRegister = (instruction as OneRegisterInstruction).registerA
+                downloadUriMethod.replaceInstruction(
+                    index,
+                    """
+                        const-string v$overrideRegister, "$downloadPath"
+                    """
+                )
+            }
+            if (instruction.opcode == Opcode.CONST_STRING) {
+                val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+                if (string.contains("/Camera")) {
+                    val overrideRegister = (instruction as OneRegisterInstruction).registerA
+                    val overrideString = string.replace("/Camera", "")
+                    downloadUriMethod.replaceInstruction(
+                        index,
+                        """
+                            const-string v$overrideRegister, "$overrideString"
+                        """
+                    )
+                }
+            }
+        }
         return PatchResultSuccess()
     }
 
+    companion object : OptionsContainer() {
+        private var downloadPathParent: String? by option(
+            PatchOption.StringListOption(
+                key = "downloadPathParent",
+                default = "DCIM",
+                options = listOf(
+                    "DCIM", "Movies", "Pictures"
+                ),
+                title = "Download Path Parent",
+                description = "Parent media directory for downloads.",
+                required = true
+            )
+        )
+        private var downloadPathChild: String? by option(
+            PatchOption.StringOption(
+                key = "downloadPathChild",
+                default = "TikTok",
+                title = "Download Path Child",
+                description = "Custom child directory name where downloaded TikTok's will be saved.",
+                required = true
+            )
+        )
```

Comment:
Is it necessary to have two separate options to specify the path? Can those be consolidated to a `downloadPath` option?

Reply:
@Sculas suggested use `StringListOption` because TikTok can only save Videos to "DCIM", "Movies" or "Pictures". If those were consolidated to a `downloadPath` , user possibly set download path to somethings like Download/Tiktok and videos can not be downloaded.